### ISSUE TITLE
[In Portal] EventGrid Trigger execution failed with NotImplementedException

### ISSUE THREAD

**Issue by shibayan** [Azure/azure-functions-host#6328]
### Repro steps

1. Create new Function App with PowerShell Core stack
2. Add new function in portal
3. Select `EventGrid Trigger` function template
4. Confirm `Microsoft.Azure.WebJobs.Extensions.EventGrid` installing dialog
5. Select created function and click `Run` button

### Error log

```
2020-01-07T16:59:11  Welcome, you are now connected to log-streaming service. The default timeout is 2 hours. Change the timeout with the App Setting SCM_LOGSTREAM_TIMEOUT (in seconds). 
2020-01-07T16:59:20.395 [Information] Executing 'Functions.EventGridTrigger1' (Reason='This function was programmatically called via the host APIs.', Id=5fb87d7b-9fa5-4ab2-a2ec-cb7f24c6621e)
2020-01-07T16:59:20.515 [Error] Executed 'Functions.EventGridTrigger1' (Failed, Id=5fb87d7b-9fa5-4ab2-a2ec-cb7f24c6621e)
The method or operation is not implemented.
```

### Addtional information

- Runtime version: 2.0.12888.0 (~2)
- Platform: Windows
- ASP Tier: Consumption
- Region: West US 2

### Workaround

Downgrade `Microsoft.Azure.WebJobs.Extensions.EventGrid` version `2.1.0` to `2.0.0`

**Comment by angelcervera** [comment]
I'm testing with different versions, always working from the Portal to be sure that I'm not doing something wrong. I'm using the nodejs code (to be able to edit from the Portal) from the Documentation.

I think that the problem is the combination between **Runtime Version and Extension Bundle Version**, but I need more time to find which combination is worng.

At this moment, I was not able to get free time to continue with the research and I moved to Service Bus as a temporal workaround.

**Comment by angelcervera** [comment]
@AnatoliB This is my test at the moment:

Runtime | Extension Bundle | Working|Notes
------------ | ------------- | ---------|-----
2.0.13900.0 | 2.0.0 | WORKING|
3.0.13901.0 | 2.0.0 | NOT WORKING| This is the default configuration using the Portal

The question is, does a runtime version require a specific Extension Bundle version? There is a little mess in the documentation about the relationship between both.

If Extension 3 is necessary for Runtime 3, the default app created using Portal is wrong, so maybe this is the issue in my case. This is the code generated with the defaul settings:

- Application Settings: `FUNCTIONS_EXTENSION_VERSION=~3` but the host.json file generated added extension version 2.0.0 as the next point shows.
- host.json file content:
```
{
  "version": "2.0",
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[1.*, 2.0.0)"
  }
}
```
- Runtime version: 3.0.13901.0 (~3)

So `Microsoft.Azure.Functions.ExtensionBundle` is 2.0.0 but `FUNCTIONS_EXTENSION_VERSION` is 3.
**Maybe I'm confusing because naming as well.** `FUNCTIONS_EXTENSION_VERSION` is used to specify the `Runtime`, and the `Function Extension Bundle` is in the one in `host.json` file???

**Comment by fabiocav** [comment]
@AnatoliB thank you for transferring the issue over. We'll take a closer look at this to understand if this is something happening in the host if if there's an issue with the Event Grid extension itself. Based on what is described above, this is likely an issue with the extension not properly handling the invoke call made by the portal.

**Comment by ddobric** [comment]
Is there anythin gnew regarding this issue?

**Comment by fabiocav** [comment]
@ThejaChoudary / @Kotasudhakarreddy I see this is currently assigned to you. Are you working on repro and additional details or should we have t his triaged/assigned?

**Comment by sspaniel** [comment]
> @angelcervera Does this workaround help you?
> 
> > Downgrade Microsoft.Azure.WebJobs.Extensions.EventGrid version 2.1.0 to 2.0.0

This fixed the issue for me

**Comment by simonjduff** [comment]
I've just been hit by the same thing.
I have a JavaScript function integrated to Event Grid. A manual test from the portal results in a `202 Accepted` and a message in the console:
```
(Failed, Id=0feb9368-20cf-4905-ba7a-0cc392c5a035, Duration=1ms)The method or operation is not implemented.
```


Host.json contains
```
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      }
    }
  },
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[1.*, 2.0.0)"
  }
}

```

**Comment by erasmosoares** [comment]
I have the same issue!

**Comment by nitmatgeo** [comment]
In my case, I released the error was when there was an exception and when my final block was uncommented.

`  Push-OutputBinding -Name Response -Value ([HttpResponseContext]@{
             StatusCode = "xyz"
             Body       = $body
         })
`
Is it your case too? Any way to handle this?

**Comment by v-anvari** [comment]
Hi @shibayan , Please let us know if you are still seeing this issue

**Comment by gresockj** [comment]
I am still seeing issues with this.  This is happening with the following `host.json`, and Java Function runtime version ~4:
```
{
  "version": "2.0",
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[3.3.0, 4.0.0)"
  } 
}
```

**Comment by codeshinobi** [comment]
Hi. Any resolution to this? I am still getting this

**Comment by gresockj** [comment]
Also, I see in the workaround to "Downgrade Microsoft.Azure.WebJobs.Extensions.EventGrid version 2.1.0 to 2.0.0".  How do I do that using the Java library?  I'm using the azure-functions-maven-plugin.

**Comment by v-anvari** [comment]
Please open an issue here for further insights using java  - https://github.com/Azure/azure-functions-Java-worker

**Comment by htuomas** [comment]
I have this same problem with a C# function, using Microsoft.Azure.WebJobs.Extensions.EventGrid version 2.1.0. It works fine when invoked with Postman.

**Comment by oleksandr-ivanov-ah** [comment]
I also have the same issue my host.json

```
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[3.3.0, 4.0.0)"
  }
```

My os: MacOS and my runtime  
```
Azure Functions Core Tools
Core Tools Version:       4.0.4544 Commit hash: N/A  (64-bit)
Function Runtime Version: 4.3.2.18186
```

**Comment by codeshinobi** [comment]
Hi

I ended up downgrading. It was a compatibility issue.

Make sure your versions are compatible.

**Comment by MichaelVlcek** [comment]
Hey, 

In-portal, the Powershell and .NET function has the same issue. Moreover, the runtime version ~4 does not support ExtensionBundle <=2.0.0 anymore.
Runtime version ~3 does support the ExtensionBundle <=2.0.0 version, but doesn't work.
Selecting runtime version ~2 does not fix the issue either.

No combination of runtime version (~2.0, ~2, ~3, ~4) and ExtensionBundle version [1.*, 2.0.0), [2.*, 3.0.0), [3.*, 4.0.0) worked for me.

It is easy to reproduce with the default setup of Azure Function with Event Grid trigger, could someone please look into this?

**Comment by fabricioferreira** [comment]
I was able to solve the issue by following the steps from this article: https://www.c-sharpcorner.com/article/how-to-debug-azure-event-grid-trigger-function-using-postman/

TL;DR;
Add the `aeg-event-type` header with value `Notification` to the request.

I was able to test it using Postman.
